components/concept/storage/src/main/java/mozilla/components/concept/storage/BookmarksStorage.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.storage
+
+import mozilla.appservices.places.BookmarkFolder
+import mozilla.appservices.places.BookmarkItem
+import mozilla.appservices.places.BookmarkRoot
+import mozilla.appservices.places.BookmarkTreeNode
+import mozilla.appservices.places.BookmarkUpdateInfo
+
+interface BookmarksStorage {
+
+    /**
+     * Produces a populated bookmarks tree for the given root.
```

Comment:
"populated" is an internal term used by the places lib with a very specific meaning (`children` array has real data). In context of an abstract interface, that term doesn't mean anything - especially since you're not defining these data types in this file. Either don't use that term, or expand it into a specific definition of what it means.